### PR TITLE
Diffusion Transformer Training Pipeline

### PR DESCRIPTION
**Pull request by zpx01**
# What does this PR do ?

Implements end-to-end diffusion transformer (DiT) pretraining / fine-tuning. 

**Collection**: diffusion

# Changelog 
- Adds DiT model implementation with cross attention functionality.
- Adds EDM diffusion sampler for higher-order sampling.
- Adds training script to train DiT models on text/image datasets.

# Usage
Readme contains instructions on how to launch training.

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [X] Did you write any new necessary tests?
- [X] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [X] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by ethanhe42** [comment]
can you fix "
Isort and Black Formatting / reformat_with_isort_and_black (pull_request_target) "?

**Comment by zpx01** [comment]
> can you fix " Isort and Black Formatting / reformat_with_isort_and_black (pull_request_target) "?

@ethanhe42 this is fixed now, it should be ready to merge.

**Comment by ethanhe42** [comment]
seems that one is still failing "Code scanning results / CodeQL"

**Review by ethanhe42 — APPROVED** [review]
